auto_backup/models/db_backup.py

```diff
@@ -186,7 +186,7 @@ def action_backup(self):
                         with rec.sftp_connection() as remote:
                             # Directory must exist
                             try:
-                                remote.makedirs(rec.folder)
+                                remote.makedirs(rec.folder, exist_ok=True)
```

Comment:
```suggestion
                                remote.makedirs(rec.folder)
```
Looks like a `pysftp.Connection` does not have this parameter 
![image](https://github.com/OCA/server-tools/assets/143788994/9421952b-1773-49c8-8995-04a0f93c4a8c)

(from https://pysftp.readthedocs.io/en/release_0.2.9/pysftp.html#pysftp.Connection.makedirs)